pyiceberg/table/snapshots.py

```diff
@@ -306,6 +307,8 @@ def build(self) -> Dict[str, str]:
         changed_partitions_size = len(self.partition_metrics)
         set_when_positive(properties, changed_partitions_size, CHANGED_PARTITION_COUNT_PROP)
         if changed_partitions_size <= self.max_changed_partitions_for_summaries:
```

Comment:
i want to point out that java also gate on the [`trustPartitionMetrics`](https://github.com/apache/iceberg/blob/07afd84080b323c37cb11cbc657fb719891ed91c/core/src/main/java/org/apache/iceberg/SnapshotSummary.java#L202) variable